compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -670,6 +670,9 @@ object TreeChecker {
       else assert(tpt.typeOpt =:= pt)
 
       // Check that the types of the args conform to the types of the contents of the hole
+      val typeArgsTypes = args.collect { case arg if arg.isType =>
```

Comment:
Will the new checker still work with tasty files generated by previous compiler versions?

Reply:
TASTy files will not contain Holes so it will work.

Pickled quotes will not contain `content` and we should not run this check on them. I will add a guard.